src/Models/User.php

```diff
@@ -0,0 +1,227 @@
+        $site_memberships = $this->getSiteMemberships()->all();
+        $sites = array_combine(
+            array_map(
+                function ($membership) {
+                    return $membership->site->id;
+                },
+                $site_memberships
+            ),
+            array_map(
+                function ($membership) {
+                    return $membership->site;
+                },
+                $site_memberships
+            )
+        );
+        return $sites;
+    }
+
+    /**
+     * Formats User object into an associative array for output
+     *
+     * @return array $data associative array of data for output
+     */
+    public function serialize()
+    {
+        $first_name = $last_name = null;
+        if (isset($this->get('profile')->firstname)) {
+            $first_name = $this->get('profile')->firstname;
+        }
+        if (isset($this->get('profile')->lastname)) {
+            $last_name = $this->get('profile')->lastname;
+        }
+
+        $data = [
+            'firstname' => $first_name,
+            'lastname' => $last_name,
+            'email' => $this->get('email'),
+            'id' => $this->id,
+        ];
+        return $data;
+    }
+
+    /**
+     * Retrieves Drush aliases for this user
+     *
+     * @return \stdClass
+     */
+    public function getAliases()
+    {
+        if (!$this->aliases) {
+            $this->fetchAliases();
+        }
+        return $this->aliases;
+    }
+
+    /**
+     * Requests API data and populates $this->aliases
+     *
+     * @return void
+     */
+    private function fetchAliases()
+    {
+        $path = "users/{$this->id}/drush_aliases";
+        $options = ['method' => 'get',];
+        $response = $this->request->request($path, $options);
+
+        $this->aliases = $response['data']->drush_aliases;
+    }
+
+    /**
+     * @return \Terminus\Collections\Instruments
+     */
+    public function getInstruments()
+    {
+        if (empty($this->instruments)) {
+            $this->instruments = $this->getContainer()->get(Instruments::class, [['user' => $this,]]);
+        }
+        return $this->instruments;
+    }
+
+    /**
+     * @return \Terminus\Collections\Instruments
+     */
+    public function getMachineTokens()
+    {
+        if (empty($this->machine_tokens)) {
+            $this->machine_tokens = $this->getContainer()->get(MachineTokens::class, [['user' => $this,]]);
+        }
+        return $this->machine_tokens;
+    }
+
+    /**
+     * @return \Terminus\Collections\UserOrganizationMemberships
+     */
+    public function getOrgMemberships()
+    {
+        if (empty($this->org_memberships)) {
+            $this->org_memberships = $this->getContainer()
+                ->get(UserOrganizationMemberships::class, [['user' => $this,]]);
+        }
+        return $this->org_memberships;
+    }
+
+    /**
+     * @return \Terminus\Collections\UserSiteMemberships
+     */
+    public function getSiteMemberships()
```

Comment:
Well we don't need them but I do prefer them. They make testing a lot easier, and they add a layer of protection between this internals of this object and the rest of the codebase. They also allow lazy-instantiation of these collections which feels like a good thing to me since most commands do not use all of these.